packages/@aws-cdk/cloudformation-diff/lib/diff/types.ts

```diff
@@ -128,11 +128,11 @@ export class TemplateDiff implements ITemplateDiff {
         continue;
       }
 
-      if (!resourceChange.newResourceType) {
+      if (!resourceChange.resourceType) {
```

Comment:
why change from `newResourceType` to `resourceType`?

Reply:
Great question! I will leave a comment in the commit I push explaining why this is resourceType, because I think it's worth documenting in some form.

The reason is, when a resource  (let's say resource X, an AWS::IAM::Policy) is removed from a template, then X.oldResourceType == AWS::IAM::Policy, but X.newResourceType == undefined -- and we need the resourceType to be able to detect if there is a security relevant diff! (Note that X.resourceType == X.oldResourceType || X.newResourceType). However, suppose X had a policy on it stating that user 012345678901 wasn't allowed to get items from a DDB table or wasn't allowed to invoke a lambda function -- now user 012345678901 is allowed to do those things, and so the diff is security relevant.

Therefore, we should check the resourceType instead of newResourceType, which might be undefined.

Reply:
For a concrete example, here's a test case I added for this issue: https://github.com/aws/aws-cdk/blob/38f1e5ca3860ab365198d670eb5f75dab5c3e452/packages/%40aws-cdk/cloudformation-diff/test/iam/detect-changes.test.ts#L461-L482

So before this change, that test case would have failed